README.md

```diff
@@ -4,16 +4,150 @@ AtacWorks is a deep learning toolkit for track denoising and peak calling from l
+    export PATH="$PATH:<custom_path>"
+    ```
+
+* Install pip dependencies
+
+    ```
+    pip install -r requirements-pip.txt
+    ```
+## Sample installation test
+* Run this command to ensure your environment is setup correctly.
+
+## Environment setup and pre-processing of data
+* Setup environment variables
+    ```
+    example_dir=$(readlink -f $(dirname "$0"));
+    data_dir="$example_dir/data";
+    ref_dir="$example_dir/reference";
+    out_dir="$example_dir/result";
+    root_dir=$(readlink -f "$example_dir/..");
+    saved_model_dir="$root_dir/data/pretrained_models";
+    ```
+
+* Convert peak files into bigwig format
+    ```
+    # Clean peaks
+    python $root_dir/peak2bw.py \
+        $data_dir/HSC.80M.chr123.10mb.peaks.bed \
+        $ref_dir/hg19.auto.sizes \
+        --prefix=$out_dir/HSC.80M.chr123.10mb.peaks.bed
+    ```
+    ```
+    # Noisy peaks
+    python $root_dir/peak2bw.py \
+        $data_dir/HSC.5M.chr123.10mb.peaks.bed \
+        $ref_dir/hg19.auto.sizes \
+        --prefix=$out_dir/HSC.5M.chr123.10mb.peaks.bed
+    ```
+
+* Split the dataset into train, val and holdout/test intervals.
+    ```
+    python $root_dir/get_intervals.py \
+        $data_dir/example.sizes 24000 $out_dir/example \
+        --val chr2 --holdout chr3
+    ```
 
-Much more information and examples can be found in the AtacWorks preprint: https://www.biorxiv.org/content/10.1101/829481
+* Save the data in H5.
+    ```
+    # Training data
+    python $root_dir/bw2h5.py \
+        --noisybw $data_dir/HSC.5M.chr123.10mb.coverage.bw \
+        --intervals $out_dir/example.training_intervals.bed \
+        --batch_size 4 \
+        --prefix $out_dir/train_data \
+        --cleanbw $data_dir/HSC.80M.chr123.10mb.coverage.bw \
+        --cleanpeakbw $out_dir/HSC.80M.chr123.10mb.peaks.bed.bw \
+        --nonzero
+    # Validation data
+    python $root_dir/bw2h5.py \
+        --noisybw $data_dir/HSC.5M.chr123.10mb.coverage.bw \
+        --intervals $out_dir/example.val_intervals.bed \
+        --batch_size 64 \
+        --prefix $out_dir/val_data \
+        --cleanbw $data_dir/HSC.80M.chr123.10mb.coverage.bw \
+        --cleanpeakbw $out_dir/HSC.80M.chr123.10mb.peaks.bed.bw
+    # Test data
+    python $root_dir/bw2h5.py \
+        --noisybw $data_dir/HSC.5M.chr123.10mb.coverage.bw \
+        --intervals $out_dir/example.holdout_intervals.bed \
+        --batch_size 64 \
+        --prefix $out_dir/test_data \
+        --cleanbw $data_dir/HSC.80M.chr123.10mb.coverage.bw \
+        --cleanpeakbw $out_dir/HSC.80M.chr123.10mb.peaks.bed.bw
+    #No label
+    python $root_dir/bw2h5.py \
+        --noisybw $data_dir/HSC.5M.chr123.10mb.coverage.bw \
+        --intervals $out_dir/example.holdout_intervals.bed \
+        --batch_size 64 \
+        --prefix $out_dir/no_label \
+        --nolabel
+    ```
+
+## Tutorial 1 - Inference and metrics using pretrained model
+* Follow steps under [pre-processing](#Environment-setup-and-pre-processing-of-data)
```

Comment:
For inference, one does not need to follow the steps for splitting the data or encoding training and validation data.

The inference and training tutorials should have completely separate workflows.

I would also suggest we keep the tutorials in independent .md files and link to them here. Otherwise this readme becomes too long and it is hard for people to find what they want.

Reply:
Currently, I just copied the steps from example/run.sh and for inference we use the `no_label.h5` which is being generated in the pre-processing step. Is there another pre-existing test data that we can use as example ?

Reply:
The current text "Follow steps under pre-processing first." suggests that they need to follow all the steps in preprocessing - which they don't. This will be much less confusing if the two workflows are independent and in separate files and we just keep what is necessary instead of saying 'follow this step but not that one'.

I will upload datasets for this example.